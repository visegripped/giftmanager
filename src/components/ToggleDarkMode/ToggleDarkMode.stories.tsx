import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import ToggleDarkMode from './index';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'component/ToggleDarkMode',
  component: ToggleDarkMode,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof ToggleDarkMode>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof ToggleDarkMode> = (args) => <ToggleDarkMode />;

export const BasicUsage = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
BasicUsage.args = {
  options: [
    { value: 'someting', label: 'something' },
    { value: 'sometingElse', label: 'something else' },
    { value: 'sometingMore', label: 'something more' },
  ],
  onChange: (event: object, uuid: string | number) => {
    console.log(' -> uuid: ', uuid);
  },
};
