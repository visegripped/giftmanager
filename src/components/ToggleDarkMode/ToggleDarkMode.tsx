import React from 'react';
import { useThemeContext } from '../../context/ThemeContext';
import './ToggleDarkMode.css';

const ToggleDarkMode = () => {
  const { dark, toggleDark } = useThemeContext();
  const handleOnChange = () => {
    toggleDark();
  };
  return (
    <label data-testid="ToggleDarkMode">
      <input type="checkbox" name="darkMode" onChange={handleOnChange} checked={dark} />
      Enable dark mode
    </label>
  );
};

export default ToggleDarkMode;
