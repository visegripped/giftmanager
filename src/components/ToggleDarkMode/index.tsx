import ToggleDarkMode from './ToggleDarkMode';

export default ToggleDarkMode;
