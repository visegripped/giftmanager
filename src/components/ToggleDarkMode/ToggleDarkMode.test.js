import { render, screen } from '@testing-library/react';
import ToggleDarkMode from './index';

describe('ToggleDarkMode', () => {
  test('snapshot matches', () => {
    render(<ToggleDarkMode />);
    const renderedToggleDarkMode = screen.getByTestId('ToggleDarkMode');
    expect(renderedToggleDarkMode).toMatchSnapshot();
  });
});
