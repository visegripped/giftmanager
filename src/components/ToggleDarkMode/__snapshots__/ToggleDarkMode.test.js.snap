// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleDarkMode snapshot matches 1`] = `
<label
  data-testid="ToggleDarkMode"
>
  <input
    name="darkMode"
    type="checkbox"
  />
  Enable dark mode
</label>
`;
